### PR TITLE
api: fix duplicated blocks of code

### PR DESCRIPTION
**Pull request by outscale-toa**
Signed-off-by: Thiery Ouattara <thiery.ouattara@outscale.com>

### CODE REVIEW

File: api/server/graph.cc [hunk 1]
```diff
@@ -653,51 +638,20 @@ std::string Graph::nic_export(const app::Nic &nic) {
...
     *out = pg_brick_tx_bytes(nic_it->second.vhost.get());
 }
 
 void Graph::nic_config_anti_spoof(const app::Nic &nic, bool enable) {
-    if (!started) {
-        LOG_ERROR_("Graph has not been started");
-        return;
-    }
-
+    check_nic_exists(nic);
```
Comment (outscale-jju) [inline api/server/graph.cc]: What happen if nic does not exist ? This won't make the function exit ...

File: api/server/graph.cc [hunk 2]
```diff
@@ -580,25 +580,10 @@ std::string Graph::nic_add(const app::Nic &nic) {
...
     auto vni_it = app::graph.vnis_.find(nic.vni);
-    if (vni_it == app::graph.vnis_.end()) {
-        LOG_ERROR_("NIC id: " + nic.id + " in vni: " +
-            std::to_string(nic.vni) + " don't seems to exist.");
-        return;
-    }
     struct GraphVni &vni = vni_it->second;
-
     auto nic_it = vni.nics.find(nic.id);
```
Comment (outscale-jju) [inline api/server/graph.cc]: Why searching again a nic when `check_nic_exists` could give it to you.

File: api/server/graph.cc [hunk 3]
```diff
@@ -957,6 +912,27 @@ void Graph::fw_add_rule(const app::Nic &nic, const app::Rule &rule) {
     fw_reload(fw);
 }
 
+void Graph::check_nic_exists(const app::Nic &nic) {
```
Comment (outscale-jju) [inline api/server/graph.cc]: maybe have a `get_nic()` would be more useful

**Comment by outscale-mgo** [comment]
Rebase this pr on next please

File: api/server/graph.cc [hunk 4]
```diff
@@ -669,53 +692,21 @@ std::string Graph::NicExport(const app::Nic &nic) {
...
-        LOG_ERROR_("Graph has not been started");
-        return;
-    }
-    *in = *out = 0;
-
-    auto vni_it = app::graph.vnis_.find(nic.vni);
-    if (vni_it == app::graph.vnis_.end()) {
-        LOG_ERROR_("NIC id: " + nic.id + " in vni: " +
-            std::to_string(nic.vni) + " don't seems to exist.");
+    Graph::GraphNic *nic_it = GetGraphNic(nic);
```
Comment (outscale-mgo) [inline api/server/graph.cc]: s/nic_it/graph_nic/

File: api/server/graph.cc [hunk 5]
```diff
@@ -669,53 +692,21 @@ std::string Graph::NicExport(const app::Nic &nic) {
...
+    if (nic_it == NULL) {
```
Comment (outscale-mgo) [inline api/server/graph.cc]: s/nic_it/graph_nic/

File: api/server/graph.cc [hunk 6]
```diff
@@ -669,53 +692,21 @@ std::string Graph::NicExport(const app::Nic &nic) {
...
-    auto nic_it = vni.nics.find(nic.id);
-    if (nic_it == vni.nics.end()) {
-        LOG_ERROR_("NIC id: " + nic.id + " in vni: " +
-            std::to_string(nic.vni) + " don't seems to exist in branch.");
-        return;
-    }
-
-    *in = pg_brick_rx_bytes(nic_it->second.vhost.get());
-    *out = pg_brick_tx_bytes(nic_it->second.vhost.get());
+    *in = pg_brick_rx_bytes(nic_it->vhost.get());
```
Comment (outscale-mgo) [inline api/server/graph.cc]: s/nic_it/graph_nic/

File: api/server/graph.cc [hunk 7]
```diff
@@ -669,53 +692,21 @@ std::string Graph::NicExport(const app::Nic &nic) {
...
+    *out = pg_brick_tx_bytes(nic_it->vhost.get());
```
Comment (outscale-mgo) [inline api/server/graph.cc]: s/nic_it/graph_nic/

File: api/server/graph.cc [hunk 8]
```diff
@@ -669,53 +692,21 @@ std::string Graph::NicExport(const app::Nic &nic) {
...
-        return;
-    }
-    struct GraphVni &vni = vni_it->second;
-
-    auto nic_it = vni.nics.find(nic.id);
-    if (nic_it == vni.nics.end()) {
-        LOG_ERROR_("NIC id: " + nic.id + " in vni: " +
-            std::to_string(nic.vni) + " don't seems to exist in branch.");
+    Graph::GraphNic *nic_it = GetGraphNic(nic);
+    if (nic_it == NULL) {
```
Comment (outscale-mgo) [inline api/server/graph.cc]: s/nic_it/graph_nic/

File: api/server/graph.cc [hunk 9]
```diff
@@ -669,53 +692,21 @@ std::string Graph::NicExport(const app::Nic &nic) {
...
         return;
     }
 
-    BrickShrPtr &antispoof = nic_it->second.antispoof;
+    BrickShrPtr &antispoof = nic_it->antispoof;
```
Comment (outscale-mgo) [inline api/server/graph.cc]: s/nic_it/graph_nic/

**Pull request closed without merging**